### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by philberty**
We have been accepting patches via mail from developers following the normal GCC processes. Let's update our guidelines to reflect this.

### CODE REVIEW

**Review by YizhePKU — APPROVED** [review]

File: CONTRIBUTING.md
Comment (dkm) [inline CONTRIBUTING.md:32]: boRs.

Maybe you should clarify where the review is happening in this case: review on the ml, and the PR is created when it has been accepted so not a real review and sent directly for merge?

**Review by flip1995 — APPROVED** [review]

File: CONTRIBUTING.md
Comment (philberty) [reply CONTRIBUTING.md:32]: Good point. A problem with accepting patches from the mailing list is when a patch might need changes, and requires us to update the PR.

I think doing code review on GitHub is much more powerful than via mail so we could still do the review here on GitHub via the PR. So when we say yes thanks for your patches and create the PR, we should forward the link back to the submitter to say here is your code under review. If the PR needs changes we can apply the patch updates from the mailing list but please expect some delays in this process due to the manual step involved.

"Reviews for the patches will be carried out via the PR on GitHub, this will be a publicly viewable resource. The link will be forwarded back to the submitter, as a way of viewing the state of their changes and as a way to manage any review comments. If a PR needs changes, please expect some delays in the management of updating the PR, due to the manual process involved."

**Review by thomasyonug — APPROVED** [review]

File: CONTRIBUTING.md
Comment (bjorn3) [reply CONTRIBUTING.md:32]: Crazy idea: What about creating a bot that automatically creates PR's based on patches on the mailing list and updates them as new patches are submitted to the mail thread? It could also forward comments both directions, so eg if you mail quoting parts of the patch, a github review gets created for the relevant lines and as you reply to them on github a mail gets sent to the mailing list. (will probably need some buffering to avoid spam on the mailing list)

Comment (flip1995) [reply CONTRIBUTING.md:32]: git does something like this with [gitgitgadget](https://github.com/gitgitgadget/gitgitgadget). Probably overkill for this project, since git is mainly developed on the mailing list, not on GitHub, but stuff like this already exists.

**Review by CohenArthur — APPROVED** [review]
I agree with Marc's suggestion, but I think it's all good! As soon as the project gets a dedicated mailing list (as it's in the works on the gcc one), it's important to add it there too for people who might want to send patches through it

File: CONTRIBUTING.md
Comment (ojeda) [reply CONTRIBUTING.md:32]: `b4`'s implementation may also prove helpful to automate stuff around patches from mailing lists.

https://git.kernel.org/pub/scm/utils/b4/b4.git/about/

Comment (dkm) [reply CONTRIBUTING.md:32]: Looks like it's similar to https://patchwork.ozlabs.org/ ? Not exactly sure...

Comment (dkm) [reply CONTRIBUTING.md:32]: Oh, I see GCC is already there : https://patchwork.ozlabs.org/project/gcc/list/

File: CONTRIBUTING.md
Comment (CohenArthur) [inline CONTRIBUTING.md:33]:
```suggestion
You can send those patches through the gcc-rust mailing list: gcc-rust@gcc.gnu.org

```

File: CONTRIBUTING.md
Comment (philberty) [reply CONTRIBUTING.md:32]: I really like the idea of automating this, is there a way for b4 to actually have a github user to create PRs? Or did I just miss that part.

Comment (ojeda) [reply CONTRIBUTING.md:32]: AFAIK, `b4` is primarily meant as a command-line tool for maintainers. I mentioned it because you may be to reuse part of its implementation to write the bot that would then contact GitHub etc.

Related: in Rust for Linux we are also trying to adapt the kernel development workflow to GitHub :-) Our bot is https://github.com/ksquirrel. I do not plan on automating the mailing list side until we are in mainline though (and even then, it will depend on the amount of patches from the mailing list).

**Review by CohenArthur — APPROVED** [review]

**Comment by tschwinge** [comment]
I streamlined the text a bit; good to go, as far as I'm concerned.

File: CONTRIBUTING.md
Comment (ojeda) [inline CONTRIBUTING.md:17]:
```suggestion
* Patches sent to the [`gcc-rust` mailing list](https://gcc.gnu.org/mailman/listinfo/gcc-rust) are likewise welcome.
```

File: CONTRIBUTING.md
Comment (dkm) [reply CONTRIBUTING.md]: What is @ksquirrel used for ? Was looking at existing solution for better integration of ml patch and github, and only find ways to create mail from PR, not the other way round.

Comment (ojeda) [reply CONTRIBUTING.md]: The goal is to have @ksquirrel automate many things within the kernel development workflow and allow us to use GitHub. Ideally, it will do all this:
  - Review commit messages for common errors (check for signatures, formatting and common practices, detect common errors such as lkml links, etc.) -- this is already implemented and I keep adding checks when I discover something new that it can check. See example at https://github.com/Rust-for-Linux/linux/pull/264 (expand the hidden comments).
  - Pick up `Acked-by`, `Reviewed-by`, `Tested-by` tags from reviewers and then appending them at merge time etc.
  - Run the CI per-commit just before the merge.
  - Collect patches from the ML and put them in GitHub automatically.

**Pull request merged**
